### PR TITLE
realtime messaging(few bugs) and Search (started)

### PR DESCRIPTION
**Pull request by victorjzsun**

### CODE REVIEW

File: src/js/controllers/messageController.js [hunk 1]
```diff
@@ -47,21 +45,24 @@ export const controlChat = async (state,chatId=0) => {
...
 
-    chatView.renderChats(state.messages.chatData,state.user.uid,state.messages.selfPos);
+    chatView.renderChats(state.messages.chatData,state.user.uid,state.messages.selfPos,state);
     //state.messages.selfPos = chatView.renderChats(state.messages.chatData,state.user.uid);
     //console.log(state.messages.selfPos)
     
 };
 
-export const messageScreen = (state) => {
+export const messageScreen = async (state) => {
```
Comment (victorjzsun) [inline src/js/controllers/messageController.js]: Why is this function async?
```suggestion
export const messageScreen = (state) => {
```

File: src/js/models/messages.js [hunk 2]
```diff
@@ -44,4 +44,19 @@ export default class Messages{
 			alert(error);
 		}
 	}
+
+	async getSearchResults(searchContent) {
```
Comment (victorjzsun) [inline src/js/models/messages.js:48]: We should really separate contacts and messages, these are two different components...

File: src/js/controllers/messageController.js [hunk 3]
```diff
@@ -136,4 +142,19 @@ export const messageScreen = (state) => {
         }
         
     });
+
+    const searchResUI = document.querySelector('.search-results');
+
+    document.querySelector('.search-btn').addEventListener('click', async e =>  {
+        e.preventDefault();
+        const btn = e.target.closest('.search-btn');
+        console.log(btn);
```
Comment (haohao1331) [inline src/js/controllers/messageController.js:151]: not a big thing, but i think logs can be deleted when commiting just so the master branch is clean

File: src/js/models/messages.js [hunk 4]
```diff
@@ -44,4 +44,19 @@ export default class Messages{
...
+		let arr = [];
+		try{
+			await db.collection("Profiles").where("name", "==", searchContent).get().then(function(querySnapshot) {
+        		querySnapshot.forEach(function(doc) {
+            		arr.push(doc.data());
+				});
+			});
+			this.searchRes = arr;
```
Comment (haohao1331) [inline src/js/models/messages.js:56]: correct indent lines 51 to 56?

Comment (victorjzsun) [inline src/js/models/messages.js]:
```suggestion
			await db.collection("Profiles").where("name", "==", searchContent).get().then(function(querySnapshot) {
        			querySnapshot.forEach(function(doc) {
            				arr.push(doc.data());
				});
			});
			this.searchRes = arr;
```

Comment (victorjzsun) [reply src/js/models/messages.js:56]: Suggested change at https://github.com/Yuxuan-Lin/VioLegacy/pull/31#pullrequestreview-509961094 applied

File: src/js/controllers/messageController.js (hunk 3 above)
Comment (victorjzsun) [reply src/js/controllers/messageController.js:151]: We can create an issue to clean logs, but yeah, agree that we should keep pr code clean

File: src/js/models/messages.js (hunk 2 above)
Comment (victorjzsun) [reply src/js/models/messages.js:48]: Issue created to resolve comment

File: src/js/controllers/messageController.js (hunk 3 above)
Comment (victorjzsun) [reply src/js/controllers/messageController.js:151]: Added to issue https://github.com/Yuxuan-Lin/VioLegacy/issues/24

**Pull request merged**